### ISSUE TITLE
GPU can not working

### ISSUE THREAD

**Issue by yangt1013** [yaoyao-liu/meta-transfer-learning#29]
Hi，Thanks for sharing the code. I tried to reproduce your code. I encountered a problem. The environment configuration is the same as your code requirements, but every time the GPU is not working, the default CPU is always the default. Check the code and it has not changed. How was it solved at the time.
 Working environment: 2060ti, pytorch0.4, cuda9.0, pytorch gpu display output is true.
Looking forward to your reply

**Comment by yaoyao-liu** [comment]
Hi @yangt1013,

Thanks for your interest in our work. I've tested the code on nvidia v100 GPU and it works well. Could you please provide me with more infromation (e.g. error information)?

**Comment by yangt1013** [comment]
The problems is  GPU memory usage can not find working file,GPU-Util occupancy 2% Default ，the operating environment is required in your paper is satisfied.Using top inquire CPU working very well and GPU can not working。I really don’t know what the reason is, I would like to seek your help to see how to deal with it

**Comment by yaoyao-liu** [comment]
Hi @yangt1013,

If there is no error information, please provide me with the following information:
- your Nvidia driver version
- your anaconda version
- your python version
- your cudnn version
- the screenshots when you're running this repo

I guess it might because that you're using a Turing architecture GPU, but I am not sure. You may try to run this repo with a Volta or Pascal GPU with more than 6GB graphics memory if it is possible.

**Comment by yangt1013** [comment]
Nvidia driver  version: NVIDIA UNIX x86_64 Kernel Module  440.33.01  Wed Nov 13 00:00:22 UTC 2019
GCC version:  gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)
cudnn version：7.1.4
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 1
#define CUDNN_PATCHLEVEL 4
--
#define CUDNN_VERSION    (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)
 anaconda version：conda 4.5.11
 python version：3.5
I do not understand what is Turing architecture GPU，Volta or Pascal GPU ，I am using a computer with only one 6G graphics card, and there may be no way to superimpose other GPUs. This is the result that appears when the environment I am running is already running. Please help me analyze and guide

**Comment by yangt1013** [comment]
{'base_lr': 0.01,
 'dataset': 'MiniImageNet',
 'dataset_dir': './data/mini/',
 'eval_weights': None,
 'gamma': 0.5,
 'gpu': '1',
 'init_weights': None,
 'max_epoch': 100,
 'meta_label': 'exp1',
 'meta_lr1': 0.0001,
 'meta_lr2': 0.001,
 'model_type': 'ResNet',
 'num_batch': 100,
 'phase': 'pre_train',
 'pre_batch_size': 128,
 'pre_custom_momentum': 0.9,
 'pre_custom_weight_decay': 0.0005,
 'pre_gamma': 0.2,
 'pre_lr': 0.1,
 'pre_max_epoch': 110,
 'pre_step_size': 30,
 'seed': 0,
 'shot': 1,
 'step_size': 10,
 'train_query': 15,
 'update_step': 50,
 'val_query': 15,
 'way': 5}
Using gpu: 1
Using random seed.
Epoch 1, Loss=4.2901 Acc=0.0156:   1%| | 2/300 [00:48<1:4
running the repo

**Comment by yangt1013** [comment]
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 2060    Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   49C    P8     5W /  N/A |    422MiB /  5934MiB |      2%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1226      G   /usr/lib/xorg/Xorg                           252MiB |
|    0      1972      G   compiz                                       131MiB |
+-----------------------------------------------------------------------------+

**Comment by yangt1013** [comment]
top - 16:41:40 up  5:18,  1 user,  load average: 6.07, 3.99, 1.91
Tasks: 318 total,   1 running, 235 sleeping,   0 stopped,   0 zombie
%Cpu(s): 44.6 us,  6.1 sy,  0.0 ni, 49.3 id,  0.1 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem : 16300836 total,  2215104 free,  7775132 used,  6310600 buff/cache
KiB Swap: 23922684 total, 23922428 free,      256 used.  7809568 avail Mem 


13827 ty        20   0 8435192 5.415g 118980 S 600.3 34.8  25:46.91 python3

**Comment by yangt1013** [comment]
Looking forward to your reply, I think I can't solve this problem，Thanks a lot！

**Comment by yangt1013** [comment]
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01 Driver Version: 440.33.01 CUDA Version: 9.0 |
|-------------------------------+----------------------+----------------------+
| GPU Name Persistence-M| Bus-Id Disp.A | Volatile Uncorr. ECC |
| Fan Temp Perf Pwr:Usage/Cap| Memory-Usage | GPU-Util Compute M. |
|===============================+======================+======================|
| 0 GeForce RTX 2060 Off | 00000000:01:00.0 Off | N/A |
| N/A 49C P8 5W / N/A | 422MiB / 5934MiB | 2% Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes: GPU Memory |
| GPU PID Type Process name Usage |
|=============================================================================|
| 0 1226 G /usr/lib/xorg/Xorg 252MiB |
| 0 1972 G compiz 131MiB |
+-----------------------------------------------------------------------------+

**Comment by yaoyao-liu** [comment]
Hi @yangt1013,

It seems you're running this repo correctly except that it is not able to use GPU. 

Turing and Volta are the names of architectures of Nvidia GPUs. E.g., 1080ti is a Volta GPU, and 2080ti is a Turing GPU. 

As I don't have a Turing GPU, I am not able to test the code based on the same environment of yours. My suggestions are that you may try to upgrade the CUDA version to e.g., 9.2 (in conda, not in the system) or upgrade the PyTorch version to 1.2. Sorry that I am not sure whether these suggestions will work. 

If you are able to solve this problem, it would be great if you can share your solutions here to let more people know. Great thanks.

**Comment by Alchemist77** [comment]
1. Check your computer cuda ID (using  https://github.com/matterport/Mask_RCNN/issues/109)

2. Check torch can recognize cuda
import torch
print(torch.cuda.is_available())

3. change the line in main.py
parser.add_argument('--gpu', default='**0**') # GPU id 
(if you have only one gpu, usually set 0)

Then you can use GPU for training the model.

When I train the model with pre parameter (run_pre.py), your gpu needs almost 10gb (batch size 128).
